File: Algorithms/Warmup/plus-minus.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

int main(){
    int m; 
    scanf("%d",&m);
    int arr[m];
  int p=0,n=0,z=0;
    for(int arr_i = 0; arr_i < m; arr_i++){
       scanf("%d",&arr[arr_i]);
      if(arr[arr_i]>0)
        p++;
      if(arr[arr_i]<0)
        n++;
      if(arr[arr_i]==0)
        z++;
    }
  float c=m+0.0;
  float pr=p/c;
  float nr=n/c;
  float zr=z/c;
  printf("%f\n",pr);
  printf("%f\n",nr);
  printf("%f",zr);
    return 0;
}

File: Algorithms/Warmup/plus-minus.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int sp = 0, sn = 0, sz = 0;
        int arr[] = new int[n];
        for(int arr_i=0; arr_i < n; arr_i++){
            arr[arr_i] = in.nextInt();
            if(arr[arr_i] > 0)
                sp++;
            else if(arr[arr_i] < 0)
                sn++;
            else sz++;
        }
        System.out.println(sp * 1.0/n);
        System.out.println(sn * 1.0/n);
        System.out.println(sz * 1.0/n);
    }
}
